engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -139,28 +146,45 @@ public boolean incrementEndpointServerErrors(UUID endpointId, int maxServerError
          * It may or may not have been disabled already from the frontend or because of a 4xx error.
          */
         if (endpoint.isPresent() && endpoint.get().isEnabled()) {
-            if (endpoint.get().getServerErrors() + currentServerErrors > maxServerErrors) {
-                /*
-                 * The endpoint exceeded the max server errors allowed from configuration.
-                 * It is therefore disabled.
-                 */
-                String hql = "UPDATE Endpoint SET enabled = FALSE WHERE id = :id AND enabled IS TRUE";
-                int updated = entityManager.createQuery(hql)
-                        .setParameter("id", endpointId)
-                        .executeUpdate();
-                return updated > 0;
+            LocalDateTime currentTime = LocalDateTime.now(ZoneId.of("UTC"));
+
+            if (endpoint.get().getServerErrors() + currentServerErrors > engineConfig.getMaxServerErrors()) {
+                if (endpoint.get().getServerErrorsSince() != null) {
```

Comment:
With the current code, endpoints with `getServerErrors() > 0` before this is deployed will never be disabled because `getServerErrorsSince()` will be `null` and remain `null` forever.

Reply:
Thats right, thant you